src/cross_auth/social_providers/oauth.py

```
@@ -604,17 +669,22 @@ def send_token_request(self, data: dict[str, Any]) -> httpx.Response:
-        (e.g., JSON instead of query string).
+        Override this method to handle different response formats.
+
+        Raises:
+            OAuth2Exception: If the response cannot be parsed.
         """
         try:
             return OAuth2TokenEndpointResponse.model_validate_json(response.text)
         except ValidationError as e:
-            logger.error(f"Failed to parse token response: {str(e)}")
-            return None
+            logger.error("Failed to parse token response: %s", e)
```

Ruff (0.15.1)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)